encoders/firebase-encoders-reflective/src/main/java/com/google/firebase/encoders/reflective/ReflectiveObjectEncoderProvider.java

```diff
@@ -63,26 +68,48 @@ public void encode(@Nullable Object obj, @NonNull ObjectEncoderContext ctx) thro
   @NonNull
   @Override
   public <T> ObjectEncoder<T> get(@NonNull Class<T> type) {
-    Map<String, EncodingDescriptor> fields = new HashMap<>();
+    Map<FieldDescriptor, EncodingDescriptor> fields = new LinkedHashMap<>();
```

Comment:
wondering what would be the use case of `LinkedHashMap`?  are we going to assume the decoded json order to be associated with its getter methods?

Reply:
oops, that was not intended. removed